packages/devtools_app/lib/src/inspector/flutter/inspector_data_models.dart

```diff
@@ -1,16 +1,91 @@
 // Copyright 2019 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+import 'dart:math' show max;
+
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
 
 import '../../utils.dart';
 import '../diagnostics_node.dart';
 import '../enum_utils.dart';
+import 'story_of_your_layout/utils.dart';
 
 const Type boxConstraintsType = BoxConstraints;
 
+/// Compute real widget sizes into rendered sizes to be displayed on the details tab.
+/// The sum of the resulting render sizes may or may not be greater than the [maxSizeAvailable]
+/// In the case where it is greater, we should render it with scrolling capability.
+///
+/// if [forceToOccupyMaxSizeAvailable] is set to true,
+///   this method will ignore the largestRenderSize
+///   and compute it's own largestRenderSize to force
+///   the sum of the render size to be equals to [maxSpaceAvailable]
+///
+/// Formula for computing render size:
+///   rs_i = (s_i - ss) * (lrs - srs) / (ls - ss) + srs
```

Comment:
use longer names instead of abbrevations and avoid underscores. For sample smallestSize instead of ss and largestSize instead of ls. It is ok if the formula takes a couple lines.
rs_i, s_i, and ss don't mean anything obvious to someone new to the code.